### DISCUSSION TITLE
Il campo net_price è richiesto poiché use_gross_prices vale false

### DISCUSSION THREAD

**Discussion by cleoni** [fattureincloud/api#38] [Q&A] [answered]
Sto lavorando sulla trasmissione di un ordine via PHP.
Ho inserito nella chiamata i dati sui prodotti che compongono l'ordine. Il sistema mi ritorna questo errore apparentemente immotivato (v. sotto perché)

messaggio:
[422] Client error: `POST https://api-v2.fattureincloud.it/c/(nostro company id)/issued_documents` resulted in a `422 Unprocessable Entity` response:
{"error":{"message":"Il campo net_price è richiesto poiché use_gross_prices vale false.","validation_result":n (truncated...)

In realtà, i prodotti inseriti nell'ordine sono due e per entrambi il net_price c'è.
Riporto in codice che, piazzato prima della trasmissione dell'oggetto $order, crea il problema (se lo elimino, tutto funziona ottenendo un ordine senza prodotti)

		$order->setItemsList(
			array(
				new IssuedDocumentItemsListItem(
					$itemslist
				)
			)
		);

Cosa contiene $itemslist si vede da questo dump:

	Array
	(
		[0] => Array
			(
				[product_id] => 76905
				[code] => 76905 20P
				[name] => Piastre serie 80 L524 H200
				[qty] => 2
				[net_price] => 28
				[vat] => FattureInCloud\Model\VatType Object
					(
						[container:protected] => Array
							(
								[id] => 55
								[value] => 
								[description] => 
								[notes] => 
								[e_invoice] => 
								[ei_type] => 
								[ei_description] => 
								[editable] => 
								[is_disabled] => 
							)

					)

			)

		[1] => Array
			(
				[product_id] => 1
				[code] => 
				[name] => Spese di Spedizione
				[qty] => 1
				[net_price] => 27
				[vat] => FattureInCloud\Model\VatType Object
					(
						[container:protected] => Array
							(
								[id] => 55
								[value] => 
								[description] => 
								[notes] => 
								[e_invoice] => 
								[ei_type] => 
								[ei_description] => 
								[editable] => 
								[is_disabled] => 
							)

					)

			)

	)

La lista degli items è stata ottenuta seguendo un esempio sulla doc. Sicuramente avrò sbagliato qualcosa ma net_price lo vedo... vi chiedo un aiuto per capire cosa non vada.

La trasmissione dell'oggetto $order è fatta come da documentazione senza la lista degli items funziona.

Anche qua, mi rimane sempre il dubbio di cosa poi la API abbia effettivamente spedito all'end point.
C'è modo di vedere il JSON che viene effettivamente spedito?

**Comment by valmoz** [comment]
Buongiorno @cleoni 
senza conoscere il company_id, diventa difficile controllare cosa sta fallendo nel suo caso perché non posso vedere il JSON effettivo. Non lo invii qui, solitamente quando serve indagare un errore è meglio passare dal supporto clienti, che può recuperare il Company ID.

Comunque, dal dump sembrerebbe che siano stati inseriti due oggetti in $itemslist, ma IssuedDocumentItemsListItem dovrebbe rappresentare un singolo elemento.

Dovrebbe somigliare a qualcosa del tipo:
```php
$order->setItemsList(
		array(
			new IssuedDocumentItemsListItem(
				$item1
			),
                         new IssuedDocumentItemsListItem(
				$item2
			),
		)
	);
```

Per vedere il JSON inviato, può utilizzare la stessa istruzione utilizzata dall'SDK stessa.

Ad esempio, nel file [IssuedDocumentsApi.php](https://github.com/fattureincloud/fattureincloud-php-sdk/blob/master/lib/Api/IssuedDocumentsApi.php) a riga 361 può leggere:

```php
$httpBody = \GuzzleHttp\json_encode(ObjectSerializer::sanitizeForSerialization($create_issued_document_request));
```
dove $create_issued_document_request contiene la richiesta stessa.

**Reply by valmoz** [reply]
Può vedere esempi più dettagliati qui: https://developers.fattureincloud.it/docs/guides/invoice-totals/

**Comment by cleoni** [comment, marked as answer]
Problema risolto, l'items list deve essere un array di item. Io avevo fatto un array di array.

Nel caso capiti ad altri di avere dubbi, è stato risolutivo stampare la richiesta prima della chiamata all'invio:

			echo "create_issued_document_request:<pre>".print_r($create_issued_document_request,1)."</pre>";
			$result = $apiInstance->createIssuedDocument($COMPANY_ID, $create_issued_document_request);

grazie

**Comment by michelelattanzio** [comment]
Siete i miei salvatori. Stavo sclerando da ore per lo stesso problema. ;)
